src/internal/wasm.ts

```diff
@@ -35,28 +39,42 @@ export const readResponse = (pointer: number, buffer: Uint8Array) => {
 };
 
 export const runWasm = async ({
-  wasmBytes,
-  wasmFunction,
+  step,
   data,
 }: {
-  wasmBytes: Uint8Array;
-  wasmFunction: string;
+  step: PipelineStep;
   data: Uint8Array;
 }) => {
-  const wasm = await WebAssembly.compile(wasmBytes);
+  if (!step.WasmBytes || !step.WasmFunction) {
+    throw Error(`No wasm function found for step ${step.name}`);
+  }
+
+  const wasm = await WebAssembly.compile(step.WasmBytes);
   const importObject = { wasi_snapshot_preview1: wasi.wasiImport };
   const instance: any = await WebAssembly.instantiate(wasm, importObject);
   const { exports } = instance;
-  const { memory, alloc, [wasmFunction]: f } = exports;
+  const { memory, alloc, [step.WasmFunction]: f } = exports;
+
+  const request = WASMRequest.create({
```

Comment:
Previously I wasn't properly creating the wasm request...I was just passing the raw data to the function.